### ISSUE TITLE
[BUG] Eslint throwing error with tsx

### ISSUE THREAD

**Issue by vatsalsinghkv** [vatsalsinghkv/easy-fix#42]
### Description

`main.tsx` is showing eslint errors.

### Screenshots

<img width="897" alt="Screenshot 2023-10-06 at 11 20 16 PM" src="https://github.com/vatsalsinghkv/easy-fix/assets/68834718/02408994-b566-4828-8be3-355c1f6694f9">

### ✨ Browser

Brave

### Checklist 🚀

- [X] I checked and didn't find similar [issue](https://github.com/vatsalsinghkv/easy-fix/issues)
- [X] I have read the [Contributing Guidelines](https://github.com/vatsalsinghkv/easy-fix/blob/main/CONTRIBUTING.md)
- [ ] I am willing to work on this issue (blank for no).

**Comment by gandholi-kumar** [comment]
Hi @vatsalsinghkv 
I would like to work on this issue. Could you kindly assign it to me?

**Query:** 
Do we need the "!" mark ?

**Comment by fekete965** [comment]
> Hi @vatsalsinghkv I would like to work on this issue. Could you kindly assign it to me?
> 
> **Query:** Do we need the "!" mark ?

Technically it is "safe" to use `non-null assertion` in this scenario, but ideally, we should throw an error with a meaningful message to the developer if the HTML node is missing.

**Comment by DhairyaMajmudar** [comment]
@vatsalsinghkv I would like to work on this issue. Can you pls. Assign this issue to me. My approach will be doing rew. Changes in the config files and main.tsx file

**Comment by vatsalsinghkv** [comment]
> Hi @vatsalsinghkv I would like to work on this issue. Could you kindly assign it to me?
> 
> **Query:** Do we need the "!" mark ?

I've used `!` assuming `#root` will exist in DOM this will throw an unhandled error if it couldn't find `#div`. If I remove`!` then I have to write a fallback code for the case what if `#root` doesn't exists in the DOM, which doesn't make sense as we are using react there must be `#root`

**Comment by gandholi-kumar** [comment]
Hi @vatsalsinghkv 
Due to some unavoidable reasons, I'm not able to work on this task, can you kindly assign it to @DhairyaMajmudar.
Sorry for the inconvenience.

**Comment by fekete965** [comment]
> @vatsalsinghkv I would like to work on this issue. Can you pls. Assign this issue to me. My approach will be doing rew. Changes in the config files and main.tsx file

I don't think we should change the eslint to allow this feature. It is better to avoid using `non-null assertion`.

**Comment by vatsalsinghkv** [comment]
@fekete965 actually null assertion for root element is kinda standard

**Comment by fekete965** [comment]
> @fekete965 actually null assertion for root element is kinda standard, basically we need to add tslint

You cannot be 100% sure the root element is there IMO. You don't create your root element but rely on a static HTML file which can be tinkered with. It would be nicer to throw an error to the console instead of letting the application crash.

TSlint is deprecated, it shouldn't be used, `typescript-eslint` should be added instead

**Comment by DhairyaMajmudar** [comment]
> Hi @vatsalsinghkv Due to some unavoidable reasons, I'm not able to work on this task, can you kindly assign it to @DhairyaMajmudar. Sorry for the inconvenience.

@vatsalsinghkv I would be pleased if this issue is assigned to me.
